spec_integration/facter_resolve_spec.rb

```diff
@@ -3,6 +3,27 @@
 require_relative 'spec_helper'
 
 describe Facter do
+  context 'when recursively calling facter in an external/custom fact' do
+    # this will need to be wrapped into a method that will install a posix or windows custom fact
```

Comment:
@joshcooper I am expecting to have to refactor this part here to create an executable fact on windows; are there any examples or documentation you can point me to?

Reply:
@tvpartytonight I think facter on Windows solely relies on the file extension (instead of file mode) to determine if it's executable or not. I'd suggest creating a bat file like:

```ruby
ext_fact_path = File.join(ext_dir, 'extfact.bat')

File.write(ext_fact_path, <<~SCRIPTEND)
facter --external-dir #{ext_dir} os
SCRIPTEND
```

One thing to watch out for is (I think) `ext_dir` will need to contain backslashes when used inside a `bat` file/ So you may need to do something like:

```ruby
ext_fact_path.tr!('/', '\\')
```

Reply:
oh, hmm, I am surprised you can just call facter here in the test without specifying which facter to load, but it seems to work just fine on my macos box too.